crates/astria-sequencer-relayer/src/relayer/mod.rs

```diff
@@ -273,13 +296,63 @@ impl Relayer {
         }
         Ok(())
     }
+
+    async fn determine_last_submitted_sequencer_height(
+        &self,
+    ) -> eyre::Result<Option<SequencerHeight>> {
+        let presubmit_path = self.presubmit_path.clone();
+        let presubmit_state = crate::utils::flatten(
+            tokio::task::spawn_blocking(move || read_state_file(presubmit_path)).await,
+        )
+        .wrap_err("failed reading presubmit state from provided file")?;
+        let postsubmit_path = self.postsubmit_path.clone();
+        let postsubmit_state = crate::utils::flatten(
+            tokio::task::spawn_blocking(move || read_state_file(postsubmit_path)).await,
+        )
+        .wrap_err("failed reading postsubmit state from provided file")?;
+
+        match (presubmit_state, postsubmit_state) {
+            (None, None) => Ok(None),
+            (Some(pre), Some(post)) if write::are_states_consistent(pre, post) => {
+                let determined_height = post.sequencer_height();
+                let height = determined_height.try_into().wrap_err_with(|| {
+                    format!(
+                        "last submitted sequencer height was determined to be \
+                         `{determined_height}`, but couldn't be turned into cometbft/sequencer \
+                         height"
+                    )
```

Comment:
Is this comment correct? Wouldn't another primary these get out of sync be that crashed while sending the data to celestia? IE submission started but response not received, so no clear idea of if got submitted or not?

Reply:
This line is just trying to convert a u64 to a i64. The if-guard of the match arm handles consistency - `write::are_states_consistent`

Reply:
No longer relevant: the states now contain `tendermint::block::Height`s so this is ensured at deserialization time (which is what it should have been).